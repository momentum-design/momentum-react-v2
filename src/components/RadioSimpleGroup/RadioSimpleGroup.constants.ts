const CLASS_PREFIX = 'md-radio-simple-group';

const STYLE = {
  wrapper: `${CLASS_PREFIX}-wrapper`,
};

export { CLASS_PREFIX, STYLE };
