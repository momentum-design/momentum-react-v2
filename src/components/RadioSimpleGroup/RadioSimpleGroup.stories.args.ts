import { commonStyles } from '../../storybook/helper.stories.argtypes';

const radioSimpleGroupArgTypes = {
  children: {
    description:
      'The children is used to render the RadioSimple components inside the RadioSimpleGroup.',
    control: { type: 'ReactNode', required: true },
    table: {
      type: {
        summary: 'ReactNode',
      },
      defaultValue: undefined,
    },
  },
  description: {
    description:
      'The description is used to provide additional context to the user.',
    control: { type: 'ReactNode', required: false },
    table: {
      type: {
        summary: 'ReactNode',
      },
      defaultValue: undefined,
    },
  },
  label: {
    description:
      'The label is used to provide a title for the group of RadioSimpleGroup.',
    control: { type: 'ReactNode', required: false },
    table: {
      type: {
        summary: 'ReactNode',
      },
      defaultValue: undefined,
    },
  },
  value: {
    description:
      'The current value for the RadioSimpleGroup.',
    control: { type: 'text', required: false },
    table: {
      type: {
        summary: 'string',
      },
      defaultValue: undefined,
    },
  },
  defaultValue: {
    description:
      'The default value for the RadioSimpleGroup.',
    control: { type: 'text', required: false },
    table: {
      type: {
        summary: 'string',
      },
      defaultValue: undefined,
    },
  },
  isDisabled: {
    description: 'Whether the group input is disabled.',
    control: { type: 'boolean' },
    table: {
      type: {
        summary: 'boolean',
      },
    },
    defaultValue: false,
  },
  isReadOnly: {
    description: 'Whether the input can be selected but not changed by the user.',
    control: { type: 'boolean' },
    table: {
      type: {
        summary: 'boolean',
      },
    },
  },
};

export { radioSimpleGroupArgTypes };

export default {
  ...commonStyles,
  ...radioSimpleGroupArgTypes,
};
