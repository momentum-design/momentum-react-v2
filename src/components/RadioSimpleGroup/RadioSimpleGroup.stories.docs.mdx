The `<RadioSimpleGroup />` component.

- Passing reactNode using `RadioSimple` component to the children.

Extended `AriaRadioProps`. See
[React aria useRadioGroup](https://react-spectrum.adobe.com/react-aria/useRadioGroup.html)
