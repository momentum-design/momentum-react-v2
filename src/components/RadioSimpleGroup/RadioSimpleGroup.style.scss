.md-radio-simple-group-wrapper {
  display: flex;
  justify-content: start;
  width: fit-content;
  flex-direction: column;
}
