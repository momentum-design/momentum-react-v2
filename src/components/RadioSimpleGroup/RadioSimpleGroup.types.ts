import { CSSProperties, ReactNode } from 'react';
import { AriaRadioGroupProps } from '@react-types/radio';

export interface RadioSimpleGroupProps extends AriaRadioGroupProps {
  /**
   * Custom class for overriding this component's CSS.
   */
  className?: string;

  /**
   * A wrapper that contains the RadioSimples inside this RadioSimpleGroup
   */
  children: ReactNode;

  /**
   * The optional description for the RadioGroup.
   */
  description?: string;

  /**
   * The optional visible RadioSimpleGroup label
   */
  label?: ReactNode;

  /**
   * Custom style for overriding this component's CSS.
   */
  style?: CSSProperties;
}
