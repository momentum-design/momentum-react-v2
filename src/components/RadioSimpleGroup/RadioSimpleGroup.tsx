import React, { FC } from 'react';
import classnames from 'classnames';
import { useRadioGroup } from '@react-aria/radio';
import { useRadioGroupState } from '@react-stately/radio';

import { STYLE } from './RadioSimpleGroup.constants';
import { RadioSimpleGroupProps } from './RadioSimpleGroup.types';
import './RadioSimpleGroup.style.scss';

/**
 * The RadioSimpleGroup component.
 */
export const RadioSimpleGroupContext = React.createContext(null);

const RadioSimpleGroup: FC<RadioSimpleGroupProps> = (props: RadioSimpleGroupProps) => {
  const { className, description, id, label, children, style } = props;

  const state = useRadioGroupState(props);
  const { radioGroupProps, labelProps } = useRadioGroup(props, state);

  return (
    <div
      {...radioGroupProps}
      className={classnames(className, STYLE.wrapper)}
      data-disabled={state.isDisabled}
      id={id}
      style={style}
    >
      {label && <span {...labelProps}>{label}</span>}
      {description}
      <RadioSimpleGroupContext.Provider value={state}>{children}</RadioSimpleGroupContext.Provider>
    </div>
  );
};

export default RadioSimpleGroup;
