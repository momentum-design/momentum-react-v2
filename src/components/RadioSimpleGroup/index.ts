import RadioSimpleGroup from './RadioSimpleGroup';

export * as RADIO_SIMPLE_GROUP_CONSTANTS from './RadioSimpleGroup.constants';
export * from './RadioSimpleGroup.types';
export { RadioSimpleGroupContext } from './RadioSimpleGroup';
export default RadioSimpleGroup;
