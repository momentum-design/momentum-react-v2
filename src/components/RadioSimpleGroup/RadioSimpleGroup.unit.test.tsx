import React from 'react';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import RadioSimpleGroup from '.';
import { STYLE } from './RadioSimpleGroup.constants';
import RadioSimple from '../RadioSimple';
import Text from '../Text';
import Icon from '../Icon';

describe('<RadioSimpleGroup />', () => {
  const children = [
    <RadioSimple value="red" key="0">
      <Icon key="0" name="accessibility" autoScale/>
      <Text key="1">Red</Text>
    </RadioSimple>,
    <RadioSimple value="blue" key="1">
      <Icon key="0" name="search" autoScale/>
      <Text key="1">Blue</Text>
    </RadioSimple>,
    <RadioSimple value="yellow" key="2">
      <Icon key="0" name="accessories" autoScale/>
      <Text key="1">Yellow</Text>
    </RadioSimple>
  ];
  const label = (<div>Select your favorite color</div>);
  const description = (<div>Only choose one</div>);

  describe('snapshot', () => {
    it('should match snapshot', () => {
      expect.assertions(1);

      const { asFragment } = render(<RadioSimpleGroup children={children} label={label} description={description} />);

      expect(asFragment()).toMatchSnapshot();
    });

    it('should match snapshot with children', () => {
      expect.assertions(1);

      const { asFragment } = render(
        <RadioSimpleGroup children={children} />
      );

      expect(asFragment()).toMatchSnapshot();
    });

    it('should match snapshot with className', () => {
      expect.assertions(1);

      const className = 'example-className';

      const { asFragment } = render(<RadioSimpleGroup children={children} className={className} />);

      expect(asFragment()).toMatchSnapshot();
    });

    it('should match snapshot with id', () => {
      expect.assertions(1);

      const id = 'example-id';

      const { asFragment } = render(<RadioSimpleGroup children={children} id={id} />);

      expect(asFragment()).toMatchSnapshot();
    });

    it('should match snapshot with style', () => {
      expect.assertions(1);

      const style = { color: 'pink' };

      const { asFragment } = render(<RadioSimpleGroup children={children} style={style} />);

      expect(asFragment()).toMatchSnapshot();
    });

    it('should match snapshot with description', () => {
      expect.assertions(1);

      const { asFragment } = render(
        <RadioSimpleGroup
          children={children}
          description={description}
        />
      );

      expect(asFragment()).toMatchSnapshot();
    });

    it('should match snapshot with child thats selected', () => {
      expect.assertions(1);

      const { asFragment } = render(
        <RadioSimpleGroup
          children={children}
          defaultValue="red"
        />
      );

      expect(asFragment()).toMatchSnapshot();
    });

    it('should match snapshot with children with group disabled', () => {
      expect.assertions(1);

      const { asFragment } = render(
        <RadioSimpleGroup
          children={children}
          isDisabled={true}
        />
      );

      expect(asFragment()).toMatchSnapshot();
    });

    it('should match snapshot with children with read only', () => {
      expect.assertions(1);

      const { asFragment } = render(
        <RadioSimpleGroup
          children={children}
          isReadOnly={true}
        />
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('attributes', () => {
    it('should have its default wrapper class', () => {
      expect.assertions(1);

      render(<RadioSimpleGroup children={children} label={label} description={description} />);

      const radioGroup = screen.getByRole('radiogroup');

      expect(radioGroup).toHaveClass(STYLE.wrapper);
    });

    it('should have provided class when className is provided', () => {
      expect.assertions(1);

      const className = 'example-className';

      render(<RadioSimpleGroup children={children} className={className} />);

      const radioGroup = screen.getByRole('radiogroup');

      expect(radioGroup).toHaveClass(className);
    });

    it('should have provided id when id is provided', () => {
      expect.assertions(1);

      const id = 'example-id';

      render(<RadioSimpleGroup children={children} id={id} />);

      const radioGroup = screen.getByRole('radiogroup');

      expect(radioGroup.id).toBe(id);
    });

    it('should have provided style when style is provided', () => {
      expect.assertions(1);

      const style = { color: 'pink' };
      const styleString = 'color: pink;';

      render(<RadioSimpleGroup children={children} style={style} />);

      const radioGroup = screen.getByRole('radiogroup');

      expect(radioGroup).toHaveStyle(styleString);
    });

    it('should have provided isReadOnly when isReadOnly is provided', () => {
      expect.assertions(1);

      render(<RadioSimpleGroup children={children} isReadOnly={true} />);

      const radioGroup = screen.getByRole('radiogroup');

      expect(radioGroup).toHaveAttribute('aria-readonly', 'true');
    });


    it('should have provided isDisabled when isDisabled is provided', () => {
      expect.assertions(1);

      render(<RadioSimpleGroup children={children} isDisabled={true} />);

      const radioGroup = screen.getByRole('radiogroup');

      expect(radioGroup).toHaveAttribute('aria-disabled', 'true');
    });

    it('should give only the select element the selected style', () => {
      expect.assertions(3);

      render(<RadioSimpleGroup children={children} defaultValue={'red'} />);

      const optionRed = screen.getByLabelText('Red');
      const optionBlue = screen.getByLabelText('Blue');
      const optionYellow = screen.getByLabelText('Yellow');

      expect(optionRed).toBeChecked();
      expect(optionBlue).not.toBeChecked();
      expect(optionYellow).not.toBeChecked();
    });
  });

  describe('actions', () => {
    it('should change which child is selected when clicked', async () => {
      expect.assertions(9);
      const user = userEvent.setup();

      render(<RadioSimpleGroup>{children}</RadioSimpleGroup>);

      const optionRed = screen.getByLabelText('Red');
      const optionBlue = screen.getByLabelText('Blue');
      const optionYellow = screen.getByLabelText('Yellow');

      await user.click(optionRed);

      expect(optionRed).toBeChecked();
      expect(optionBlue).not.toBeChecked();
      expect(optionYellow).not.toBeChecked();

      await user.click(optionBlue);

      expect(optionBlue).toBeChecked();
      expect(optionYellow).not.toBeChecked();
      expect(optionRed).not.toBeChecked();

      await user.click(optionYellow);

      expect(optionYellow).toBeChecked();
      expect(optionRed).not.toBeChecked();
      expect(optionBlue).not.toBeChecked();
    });

    it('should call onChange when child is clicked', async () => {
      expect.assertions(1);
      const user = userEvent.setup();

      const onChange = jest.fn();

      render(<RadioSimpleGroup onChange={onChange}>{children}</RadioSimpleGroup>);

      const optionRed = screen.getByLabelText('Red');

      await user.click(optionRed);

      expect(onChange).toBeCalledWith('red');
    });

    it('should not call onChange when default value is clicked first', async () => {
      expect.assertions(1);
      const user = userEvent.setup();

      const onChange = jest.fn();

      render(<RadioSimpleGroup defaultValue={'red'} onChange={onChange}>{children}</RadioSimpleGroup>);

      const optionRed = screen.getByText('Red');

      await user.click(optionRed);

      expect(onChange).not.toBeCalled();
    });

    it('should call onChange only once when a child is clicked several times', async () => {
      expect.assertions(2);
      const user = userEvent.setup();

      const onChange = jest.fn();

      render(<RadioSimpleGroup onChange={onChange}>{children}</RadioSimpleGroup>);

      const optionRed = screen.getByLabelText('Red');

      await user.click(optionRed);

      expect(onChange).toBeCalledWith('red');
      expect(onChange).toBeCalledTimes(1);
    });

    it('should not call onChange when the group is readonly and a child is clicked', async () => {
      expect.assertions(1);
      const user = userEvent.setup();

      const onChange = jest.fn();

      render(<RadioSimpleGroup isReadOnly={true} onChange={onChange}>{children}</RadioSimpleGroup>);

      const optionRed = screen.getByText('Red');

      await user.click(optionRed);

      expect(onChange).not.toBeCalled();
    });

    it('should not call onChange when the group is disabled and a child is clicked', async () => {
      expect.assertions(1);
      const user = userEvent.setup();

      const onChange = jest.fn();

      render(<RadioSimpleGroup isDisabled={true} onChange={onChange}>{children}</RadioSimpleGroup>);

      const optionRed = screen.getByText('Red');

      await user.click(optionRed);

      expect(onChange).not.toBeCalled();
    });

    it('should not call onChange when a disabled child is clicked', async () => {
      expect.assertions(4);
      const user = userEvent.setup();

      const onChange = jest.fn();

      render(<RadioSimpleGroup onChange={onChange}>
        <RadioSimple value="red" key="0" isDisabled={true}>
          <Icon name="accessibility" autoScale/>
          <Text>Red</Text>
        </RadioSimple>,
        <RadioSimple value="blue" key="1">
          <Icon name="search" autoScale/>
          <Text>Blue</Text>
        </RadioSimple>,
        <RadioSimple value="yellow" key="2">
          <Icon name="accessories" autoScale/>
          <Text>Yellow</Text>
        </RadioSimple>
      </RadioSimpleGroup>);

      const optionRed = screen.getByLabelText('Red');

      expect(optionRed).toBeDisabled();

      await user.click(optionRed);

      expect(onChange).not.toBeCalled();

      const optionYellow = screen.getByLabelText('Yellow');

      expect(optionYellow).not.toBeDisabled();

      await user.click(optionYellow);

      expect(onChange).toBeCalled();
    });

    it('should allow keyboard navigation', async () => {
      expect.assertions(6);
      const user = userEvent.setup();

      const onChange = jest.fn();

      render(<RadioSimpleGroup onChange={onChange} defaultValue="red">{children}</RadioSimpleGroup>);

      const optionRed = screen.getByLabelText('Red');
      const optionBlue = screen.getByLabelText('Blue');
      const optionYellow = screen.getByLabelText('Yellow');

      await user.type(optionRed, '{arrowdown}');

      expect(onChange).toHaveBeenLastCalledWith('blue');

      await user.type(optionBlue, '{arrowdown}');

      expect(onChange).toHaveBeenLastCalledWith('yellow');

      await user.type(optionYellow, '{arrowdown}');

      expect(onChange).toHaveBeenLastCalledWith('red');


      await user.type(optionYellow, '{arrowup}');

      expect(onChange).toHaveBeenLastCalledWith('blue');

      await user.type(optionBlue, '{arrowup}');

      expect(onChange).toHaveBeenLastCalledWith('red');

      await user.type(optionRed, '{arrowup}');

      expect(onChange).toHaveBeenLastCalledWith('yellow');
    });

    it('should focus selected child on keyboard input', async () => {
      expect.assertions(5);
      const user = userEvent.setup();

      const onChange = jest.fn();

      render(<RadioSimpleGroup onChange={onChange}>{children}</RadioSimpleGroup>);

      const optionRed = screen.getByLabelText('Red');
      const optionBlue = screen.getByLabelText('Blue');

      await user.tab();

      expect(optionRed).toHaveFocus();
      expect(onChange).not.toBeCalled();

      await user.type(optionRed, '{arrowdown}');

      expect(optionBlue).toHaveFocus();
      expect(optionBlue).toBeChecked();

      expect(onChange).toBeCalledWith('blue');
    });
  });
});
