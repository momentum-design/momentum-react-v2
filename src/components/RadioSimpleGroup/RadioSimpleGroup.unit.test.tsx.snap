// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioSimpleGroup /> snapshot should match snapshot 1`] = `
<DocumentFragment>
  <div
    aria-labelledby="test-ID"
    aria-orientation="vertical"
    class="md-radio-simple-group-wrapper"
    data-disabled="false"
    role="radiogroup"
  >
    <span
      id="test-ID"
    >
      <div>
        Select your favorite color
      </div>
    </span>
    <small
      class="md-text-wrapper"
      data-type="body-secondary"
    >
      Only choose one
    </small>
    <label
      class="md-radio-simple-wrapper"
      data-disabled="false"
      data-focused="false"
      data-selected="false"
    >
      <div
        style="border: 0px; clip-path: inset(50%); height: 1px; margin: 0px -1px -1px 0px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
      >
        <input
          name="test-ID"
          tabindex="0"
          type="radio"
          value="red"
        />
      </div>
      <p
        class="md-text-wrapper"
        data-type="body-primary"
      >
        Red
      </p>
    </label>
    <label
      class="md-radio-simple-wrapper"
      data-disabled="false"
      data-focused="false"
      data-selected="false"
    >
      <div
        style="border: 0px; clip-path: inset(50%); height: 1px; margin: 0px -1px -1px 0px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
      >
        <input
          name="test-ID"
          tabindex="0"
          type="radio"
          value="blue"
        />
      </div>
      <p
        class="md-text-wrapper"
        data-type="body-primary"
      >
        Blue
      </p>
    </label>
    <label
      class="md-radio-simple-wrapper"
      data-disabled="false"
      data-focused="false"
      data-selected="false"
    >
      <div
        style="border: 0px; clip-path: inset(50%); height: 1px; margin: 0px -1px -1px 0px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
      >
        <input
          name="test-ID"
          tabindex="0"
          type="radio"
          value="yellow"
        />
      </div>
      <p
        class="md-text-wrapper"
        data-type="body-primary"
      >
        Yellow
      </p>
    </label>
  </div>
</DocumentFragment>
`;

exports[`<RadioSimpleGroup /> snapshot should match snapshot with child thats selected 1`] = `
<DocumentFragment>
  <div
    aria-orientation="vertical"
    class="md-radio-simple-group-wrapper"
    data-disabled="false"
    role="radiogroup"
  >
    <label
      class="md-radio-simple-wrapper"
      data-disabled="false"
      data-focused="false"
      data-selected="true"
    >
      <div
        style="border: 0px; clip-path: inset(50%); height: 1px; margin: 0px -1px -1px 0px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
      >
        <input
          checked=""
          name="test-ID"
          tabindex="0"
          type="radio"
          value="red"
        />
      </div>
      <p
        class="md-text-wrapper"
        data-type="body-primary"
      >
        Red
      </p>
    </label>
    <label
      class="md-radio-simple-wrapper"
      data-disabled="false"
      data-focused="false"
      data-selected="false"
    >
      <div
        style="border: 0px; clip-path: inset(50%); height: 1px; margin: 0px -1px -1px 0px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
      >
        <input
          name="test-ID"
          tabindex="0"
          type="radio"
          value="blue"
        />
      </div>
      <p
        class="md-text-wrapper"
        data-type="body-primary"
      >
        Blue
      </p>
    </label>
    <label
      class="md-radio-simple-wrapper"
      data-disabled="false"
      data-focused="false"
      data-selected="false"
    >
      <div
        style="border: 0px; clip-path: inset(50%); height: 1px; margin: 0px -1px -1px 0px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
      >
        <input
          name="test-ID"
          tabindex="0"
          type="radio"
          value="yellow"
        />
      </div>
      <p
        class="md-text-wrapper"
        data-type="body-primary"
      >
        Yellow
      </p>
    </label>
  </div>
</DocumentFragment>
`;

exports[`<RadioSimpleGroup /> snapshot should match snapshot with children 1`] = `
<DocumentFragment>
  <div
    aria-orientation="vertical"
    class="md-radio-simple-group-wrapper"
    data-disabled="false"
    role="radiogroup"
  >
    <label
      class="md-radio-simple-wrapper"
      data-disabled="false"
      data-focused="false"
      data-selected="false"
    >
      <div
        style="border: 0px; clip-path: inset(50%); height: 1px; margin: 0px -1px -1px 0px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
      >
        <input
          name="test-ID"
          tabindex="0"
          type="radio"
          value="red"
        />
      </div>
      <p
        class="md-text-wrapper"
        data-type="body-primary"
      >
        Red
      </p>
    </label>
    <label
      class="md-radio-simple-wrapper"
      data-disabled="false"
      data-focused="false"
      data-selected="false"
    >
      <div
        style="border: 0px; clip-path: inset(50%); height: 1px; margin: 0px -1px -1px 0px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
      >
        <input
          name="test-ID"
          tabindex="0"
          type="radio"
          value="blue"
        />
      </div>
      <p
        class="md-text-wrapper"
        data-type="body-primary"
      >
        Blue
      </p>
    </label>
    <label
      class="md-radio-simple-wrapper"
      data-disabled="false"
      data-focused="false"
      data-selected="false"
    >
      <div
        style="border: 0px; clip-path: inset(50%); height: 1px; margin: 0px -1px -1px 0px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
      >
        <input
          name="test-ID"
          tabindex="0"
          type="radio"
          value="yellow"
        />
      </div>
      <p
        class="md-text-wrapper"
        data-type="body-primary"
      >
        Yellow
      </p>
    </label>
  </div>
</DocumentFragment>
`;

exports[`<RadioSimpleGroup /> snapshot should match snapshot with children with group disabled 1`] = `
<DocumentFragment>
  <div
    aria-disabled="true"
    aria-orientation="vertical"
    class="md-radio-simple-group-wrapper"
    data-disabled="true"
    role="radiogroup"
  >
    <label
      class="md-radio-simple-wrapper"
      data-disabled="false"
      data-focused="false"
      data-selected="false"
    >
      <div
        style="border: 0px; clip-path: inset(50%); height: 1px; margin: 0px -1px -1px 0px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
      >
        <input
          disabled=""
          name="test-ID"
          type="radio"
          value="red"
        />
      </div>
      <p
        class="md-text-wrapper"
        data-type="body-primary"
      >
        Red
      </p>
    </label>
    <label
      class="md-radio-simple-wrapper"
      data-disabled="false"
      data-focused="false"
      data-selected="false"
    >
      <div
        style="border: 0px; clip-path: inset(50%); height: 1px; margin: 0px -1px -1px 0px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
      >
        <input
          disabled=""
          name="test-ID"
          type="radio"
          value="blue"
        />
      </div>
      <p
        class="md-text-wrapper"
        data-type="body-primary"
      >
        Blue
      </p>
    </label>
    <label
      class="md-radio-simple-wrapper"
      data-disabled="false"
      data-focused="false"
      data-selected="false"
    >
      <div
        style="border: 0px; clip-path: inset(50%); height: 1px; margin: 0px -1px -1px 0px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
      >
        <input
          disabled=""
          name="test-ID"
          type="radio"
          value="yellow"
        />
      </div>
      <p
        class="md-text-wrapper"
        data-type="body-primary"
      >
        Yellow
      </p>
    </label>
  </div>
</DocumentFragment>
`;

exports[`<RadioSimpleGroup /> snapshot should match snapshot with children with read only 1`] = `
<DocumentFragment>
  <div
    aria-orientation="vertical"
    aria-readonly="true"
    class="md-radio-simple-group-wrapper"
    data-disabled="false"
    role="radiogroup"
  >
    <label
      class="md-radio-simple-wrapper"
      data-disabled="false"
      data-focused="false"
      data-selected="false"
    >
      <div
        style="border: 0px; clip-path: inset(50%); height: 1px; margin: 0px -1px -1px 0px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
      >
        <input
          aria-readonly="true"
          name="test-ID"
          tabindex="0"
          type="radio"
          value="red"
        />
      </div>
      <p
        class="md-text-wrapper"
        data-type="body-primary"
      >
        Red
      </p>
    </label>
    <label
      class="md-radio-simple-wrapper"
      data-disabled="false"
      data-focused="false"
      data-selected="false"
    >
      <div
        style="border: 0px; clip-path: inset(50%); height: 1px; margin: 0px -1px -1px 0px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
      >
        <input
          aria-readonly="true"
          name="test-ID"
          tabindex="0"
          type="radio"
          value="blue"
        />
      </div>
      <p
        class="md-text-wrapper"
        data-type="body-primary"
      >
        Blue
      </p>
    </label>
    <label
      class="md-radio-simple-wrapper"
      data-disabled="false"
      data-focused="false"
      data-selected="false"
    >
      <div
        style="border: 0px; clip-path: inset(50%); height: 1px; margin: 0px -1px -1px 0px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
      >
        <input
          aria-readonly="true"
          name="test-ID"
          tabindex="0"
          type="radio"
          value="yellow"
        />
      </div>
      <p
        class="md-text-wrapper"
        data-type="body-primary"
      >
        Yellow
      </p>
    </label>
  </div>
</DocumentFragment>
`;

exports[`<RadioSimpleGroup /> snapshot should match snapshot with className 1`] = `
<DocumentFragment>
  <div
    aria-orientation="vertical"
    class="example-className md-radio-simple-group-wrapper"
    data-disabled="false"
    role="radiogroup"
  >
    <label
      class="md-radio-simple-wrapper"
      data-disabled="false"
      data-focused="false"
      data-selected="false"
    >
      <div
        style="border: 0px; clip-path: inset(50%); height: 1px; margin: 0px -1px -1px 0px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
      >
        <input
          name="test-ID"
          tabindex="0"
          type="radio"
          value="red"
        />
      </div>
      <p
        class="md-text-wrapper"
        data-type="body-primary"
      >
        Red
      </p>
    </label>
    <label
      class="md-radio-simple-wrapper"
      data-disabled="false"
      data-focused="false"
      data-selected="false"
    >
      <div
        style="border: 0px; clip-path: inset(50%); height: 1px; margin: 0px -1px -1px 0px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
      >
        <input
          name="test-ID"
          tabindex="0"
          type="radio"
          value="blue"
        />
      </div>
      <p
        class="md-text-wrapper"
        data-type="body-primary"
      >
        Blue
      </p>
    </label>
    <label
      class="md-radio-simple-wrapper"
      data-disabled="false"
      data-focused="false"
      data-selected="false"
    >
      <div
        style="border: 0px; clip-path: inset(50%); height: 1px; margin: 0px -1px -1px 0px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
      >
        <input
          name="test-ID"
          tabindex="0"
          type="radio"
          value="yellow"
        />
      </div>
      <p
        class="md-text-wrapper"
        data-type="body-primary"
      >
        Yellow
      </p>
    </label>
  </div>
</DocumentFragment>
`;

exports[`<RadioSimpleGroup /> snapshot should match snapshot with description 1`] = `
<DocumentFragment>
  <div
    aria-orientation="vertical"
    class="md-radio-simple-group-wrapper"
    data-disabled="false"
    role="radiogroup"
  >
    <small
      class="md-text-wrapper"
      data-type="body-secondary"
    >
      Only choose one
    </small>
    <label
      class="md-radio-simple-wrapper"
      data-disabled="false"
      data-focused="false"
      data-selected="false"
    >
      <div
        style="border: 0px; clip-path: inset(50%); height: 1px; margin: 0px -1px -1px 0px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
      >
        <input
          name="test-ID"
          tabindex="0"
          type="radio"
          value="red"
        />
      </div>
      <p
        class="md-text-wrapper"
        data-type="body-primary"
      >
        Red
      </p>
    </label>
    <label
      class="md-radio-simple-wrapper"
      data-disabled="false"
      data-focused="false"
      data-selected="false"
    >
      <div
        style="border: 0px; clip-path: inset(50%); height: 1px; margin: 0px -1px -1px 0px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
      >
        <input
          name="test-ID"
          tabindex="0"
          type="radio"
          value="blue"
        />
      </div>
      <p
        class="md-text-wrapper"
        data-type="body-primary"
      >
        Blue
      </p>
    </label>
    <label
      class="md-radio-simple-wrapper"
      data-disabled="false"
      data-focused="false"
      data-selected="false"
    >
      <div
        style="border: 0px; clip-path: inset(50%); height: 1px; margin: 0px -1px -1px 0px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
      >
        <input
          name="test-ID"
          tabindex="0"
          type="radio"
          value="yellow"
        />
      </div>
      <p
        class="md-text-wrapper"
        data-type="body-primary"
      >
        Yellow
      </p>
    </label>
  </div>
</DocumentFragment>
`;

exports[`<RadioSimpleGroup /> snapshot should match snapshot with id 1`] = `
<DocumentFragment>
  <div
    aria-orientation="vertical"
    class="md-radio-simple-group-wrapper"
    data-disabled="false"
    id="example-id"
    role="radiogroup"
  >
    <label
      class="md-radio-simple-wrapper"
      data-disabled="false"
      data-focused="false"
      data-selected="false"
    >
      <div
        style="border: 0px; clip-path: inset(50%); height: 1px; margin: 0px -1px -1px 0px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
      >
        <input
          name="test-ID"
          tabindex="0"
          type="radio"
          value="red"
        />
      </div>
      <p
        class="md-text-wrapper"
        data-type="body-primary"
      >
        Red
      </p>
    </label>
    <label
      class="md-radio-simple-wrapper"
      data-disabled="false"
      data-focused="false"
      data-selected="false"
    >
      <div
        style="border: 0px; clip-path: inset(50%); height: 1px; margin: 0px -1px -1px 0px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
      >
        <input
          name="test-ID"
          tabindex="0"
          type="radio"
          value="blue"
        />
      </div>
      <p
        class="md-text-wrapper"
        data-type="body-primary"
      >
        Blue
      </p>
    </label>
    <label
      class="md-radio-simple-wrapper"
      data-disabled="false"
      data-focused="false"
      data-selected="false"
    >
      <div
        style="border: 0px; clip-path: inset(50%); height: 1px; margin: 0px -1px -1px 0px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
      >
        <input
          name="test-ID"
          tabindex="0"
          type="radio"
          value="yellow"
        />
      </div>
      <p
        class="md-text-wrapper"
        data-type="body-primary"
      >
        Yellow
      </p>
    </label>
  </div>
</DocumentFragment>
`;

exports[`<RadioSimpleGroup /> snapshot should match snapshot with style 1`] = `
<DocumentFragment>
  <div
    aria-orientation="vertical"
    class="md-radio-simple-group-wrapper"
    data-disabled="false"
    role="radiogroup"
    style="color: pink;"
  >
    <label
      class="md-radio-simple-wrapper"
      data-disabled="false"
      data-focused="false"
      data-selected="false"
    >
      <div
        style="border: 0px; clip-path: inset(50%); height: 1px; margin: 0px -1px -1px 0px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
      >
        <input
          name="test-ID"
          tabindex="0"
          type="radio"
          value="red"
        />
      </div>
      <p
        class="md-text-wrapper"
        data-type="body-primary"
      >
        Red
      </p>
    </label>
    <label
      class="md-radio-simple-wrapper"
      data-disabled="false"
      data-focused="false"
      data-selected="false"
    >
      <div
        style="border: 0px; clip-path: inset(50%); height: 1px; margin: 0px -1px -1px 0px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
      >
        <input
          name="test-ID"
          tabindex="0"
          type="radio"
          value="blue"
        />
      </div>
      <p
        class="md-text-wrapper"
        data-type="body-primary"
      >
        Blue
      </p>
    </label>
    <label
      class="md-radio-simple-wrapper"
      data-disabled="false"
      data-focused="false"
      data-selected="false"
    >
      <div
        style="border: 0px; clip-path: inset(50%); height: 1px; margin: 0px -1px -1px 0px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
      >
        <input
          name="test-ID"
          tabindex="0"
          type="radio"
          value="yellow"
        />
      </div>
      <p
        class="md-text-wrapper"
        data-type="body-primary"
      >
        Yellow
      </p>
    </label>
  </div>
</DocumentFragment>
`;
